File: Entidades/Productos.cs
using System;
using System.Runtime.CompilerServices;

namespace Entidades
{
    public class Productos
    {
        private String idproducto; 
        private String codigo; 
        private String nombre; 
        private String descripcion; 
        private double precio; 
        private int stock; 
        private String imagen; 
        private Boolean activo; 

        public Productos()
        {
            idproducto = "no definido"; 
            codigo = "no definido"; 
            nombre = "no definido"; 
            descripcion = "no definido"; 
            precio = 0;
            stock = 0; 
            imagen = "no definido"; 
            activo = true; 
        }

        public Productos(String idproducto, String codigo, String nombre, String descripcion,
         float precio, int stock, String imagen, Boolean activo)
        {
            this.idproducto = idproducto; 
            this.codigo = codigo; 
            this.nombre = nombre; 
            this.descripcion = descripcion; 
            this.precio = precio; 
            this.stock = stock; 
            this.imagen = imagen; 
            this.activo = activo; 
        }

        public String IdProducto
        {
            get
            {
                return idproducto; 
            }
            set
            {
                idproducto = value; 
            }
        }

        public String Codigo
        {
            get
            {
                return codigo; 
            }
            set
            {
                codigo = value; 
            }
        }

        public String Nombre
        {
            get
            {
                return nombre; 
            }
            set
            {
                nombre = value; 
            }
        }
        public String Descripcion
        {
            get
            {
                return descripcion; 
            }
            set
            {
                descripcion = value; 
            }
        }

        public double Precio
        {
            get
            {
                return precio; 
            }
            set{
                precio = value; 
            }
        }

        public int Stock
        {
            get
            {
                return stock; 
            }
            set
            {
                stock = value; 
            }
        }

        public String Imagen
        {
            get
            {
                return imagen; 
            }
            set
            {
                imagen = value; 
            }
        }

        public Boolean Activo
        {
            get
            {
                return activo; 
            }
            set
            {
                activo = value; 
            }
        }
    }
}

File: FrontEnd/JavaScript/Productos.js
const host = "https://localhost:7040/api/Productos"

import { crearCelda, crearCeldaImg, crearButton} from "./funciones.js"

const mensaje = document.getElementById("lbl_mensaje"); 
mensaje.textContent = ""; 

let carrito = []; 

async function cargarProductos() { 
    const respuesta_prod = await fetch(`${host}/${"productos"}`); 

    if(!respuesta_prod.ok){
        mensaje.textContent = "No se pudo conectar a la base de datos"; 
    }

    const productos = await respuesta_prod.json(); 
    const tablacuerpo = document.getElementById("cuerpo_tabla"); 

    // tablacuerpo.innerHTML = ""; 

    productos.forEach(prod => {
        const tr = document.createElement("tr"); 

        const codigo = crearCelda(prod.codigo); 
        const nombre = crearCelda(prod.nombre); 
        const descripcion = crearCelda(prod.descripcion); 
        const precio = crearCelda("$" + prod.precio);
        const stock = crearCelda(prod.stock); 
        const imagen = crearCeldaImg(prod.imagen); 
        const tipo = crearCelda(prod.tipo)
        const btn_agregar = crearButton(" + ", ""); 

        tr.appendChild(codigo); 
        tr.appendChild(nombre); 
        tr.appendChild(tipo); 
        tr.appendChild(descripcion); 
        tr.appendChild(precio); 
        tr.appendChild(stock); 
        tr.appendChild(imagen); 
        tr.appendChild(btn_agregar); 

        tablacuerpo.appendChild(tr); 

        btn_agregar.addEventListener("click", () => {
            // agregarAlCarrito(prod); 

            const tablaseleccionados = document.getElementById("tabla_seleccionados");

            const tr = document.createElement("tr");
            const btn_eliminar = crearButton("🗑️", ""); 

            tr.appendChild(crearCelda(prod.codigo));
            tr.appendChild(crearCelda(prod.nombre));
            tr.appendChild(crearCelda(prod.descripcion));
            tr.appendChild(crearCelda(prod.precio));
            tr.appendChild(btn_eliminar); 

            tablaseleccionados.appendChild(tr); 

            btn_eliminar.addEventListener("click", () => {
                tr.remove(); 
            }); 
        }); 
    });

}

// function agregarAlCarrito(prod) {
//     carrito.push(prod);
//     renderizarCarrito();
// }

// function renderizarCarrito() {
//     const tablaseleccionados = document.getElementById("tabla_seleccionados");
//     tablaseleccionados.innerHTML = "";
    
//     let tr; 
//     let btn_eliminar;
    
//     carrito.forEach(prod => {
//         tr = document.createElement("tr");
//         btn_eliminar = crearButton("🗑️", ""); 

//         tr.appendChild(crearCelda(prod.codigo));
//         tr.appendChild(crearCelda(prod.nombre));
//         tr.appendChild(crearCelda(prod.descripcion));
//         tr.appendChild(crearCelda(prod.precio));
//         tr.appendChild(btn_eliminar); 

//         tablaseleccionados.appendChild(tr);
        
//     });
    
//     btn_eliminar.addEventListener("click", () => {
//         tr.remove(); 
//     }); 
// }

// function seleccionarProductos(){
//     const tablaseleccionados = document.getElementById("tabla_seleccionados"); 
//     tablaseleccionados.innerHTML = "";
    
//     const filas = document.querySelectorAll("#cuerpo_tabla tr");  

//     filas.forEach(fila => {
//         fila.addEventListener("click", () => {
//             const tr = document.createElement("tr");
//             const celdas = fila.querySelectorAll("td");  
 
//             tr.appendChild(crearCelda(celdas[0].textContent)); 
//             tr.appendChild(crearCelda(celdas[1].textContent)); 
//             tr.appendChild(crearCelda(celdas[3].textContent)); 
//             tr.appendChild(crearCelda(celdas[4].textContent)); 
            
//             tablaseleccionados.appendChild(tr); 
//         });
        
//     });
// }

document.addEventListener("DOMContentLoaded", async () => {
    await cargarProductos(); 
});